### PR TITLE
feat: add support to all routers

### PR DESCRIPTION
**Pull request by davidebianchi**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,7 +9,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### BREAKING CHANGES
 
-- modified Router interface by sorting addRoute arguments in a different manner: first method and then path
+- modified apirouter.Router interface
+  - sorting addRoute arguments in a different manner: first method and then path
+  - AddRoute now returns a Route object and an error
+  - added `SwaggerHandler` required method to router interface
+  - AddRoute method now take `interface{}` as handler argument instead of `HandlerFunc`. This is to allow the router to be used with other handler types. For the basic gorilla apirouter implementation, only the `func(w http.ResponseWriter, req *http.Request)` type is supported. If you need other handler types, you can create it using the `apirouter.Router` interface.
```
Comment (fredmaggiowski) [inline CHANGELOG.md:16]: I liked the interface requiring the base `HandlerFunc`, at the end of the day I think all routers/web-frameworks would have to converge on that signature; maybe one could write its own adapter 🤔

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,7 +9,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### BREAKING CHANGES
 
-- modified Router interface by sorting addRoute arguments in a different manner: first method and then path
+- modified apirouter.Router interface
+  - sorting addRoute arguments in a different manner: first method and then path
+  - AddRoute now returns a Route object and an error
+  - added `SwaggerHandler` required method to router interface
```
Comment (fredmaggiowski) [inline CHANGELOG.md:15]: what is the purpose for this method?

File: README.md [hunk 3]
```diff
@@ -11,11 +11,14 @@
...
 Generate an openapi spec dynamically based on the types used to handle request and response.
 
-It works with any router which support handler net/http HandlerFunc compatible.
+It works with any router.
 
 The routers supported out of the box are:
 
-- [gorilla-mux](https://github.com/gorilla/mux)
+- [gorilla-mux](https://github.com/gorilla/mux) with handler of type `func(w http.ResponseWriter, req *http.Request)`.
+- [echo](https://echo.labstack.com/) with handler of type `func(c echo.Context) error`.
```
Comment (fredmaggiowski) [inline README.md:19]: oh I see, not all router implement the same interface, lol

**Comment by davidebianchi** [comment]
Replaced by #94

**Pull request closed without merging**